src/main/kotlin/org/danilopianini/plagiarismdetector/analyzer/representation/token/LanguageTokenTypes.kt

```diff
@@ -0,0 +1,16 @@
+package org.danilopianini.plagiarismdetector.analyzer.representation.token
+
+/**
+ * An interface which models a container of [TokenType]s for a specific language.
+ */
+interface LanguageTokenTypes {
+    /**
+     * @return the [TokenType] associated to the given construct.
+     */
+    fun getTokenTypeBy(constructName: String): TokenType
```

Comment:
shorter names sound more idiomatic
```suggestion
    fun tokenFor(constructName: String): TokenType
```